projects/client-api/src/index.js

```diff
@@ -1303,6 +1306,71 @@ export function updateZoom(level) {
 }
 chainList.updateZoom = updateZoom;
 
+/**
+ * Subscribe to label change and exit events
+ * @function selectionUpdates
+ * @param {object} [cache] - An optional cache object (empty object) to use for caching label updates across subscriptions.
+ * @return {Subscription} A {@link Subscription} that can be used to stop reacting to label updates
+ * @example
+ * GraphistryJS(document.getElementById('viz'))
+ *     .pipe(
+ *          selectionUpdates,
+ *          tap((s) => {
+ *                 console.log(s);
+ *          }),
+ *     })
+ *     .subscribe();
+ */
+export function selectionUpdates(g) {
+    console.log('client-api.APISUB', 'selectionUpdates', g, 'exor');
+    if (!(g.subscriptionAPIVersion >= 1)) {
+        throw new Error('selectionUpdates is not available the currently embedded graphistry viz.');
+    }
+
+    const SELECTION_PATH = "workbooks.open.views.current.selection['edge','point']";
+    return g.selectionStream || (g.selectionStream = new BehaviorSubject('value'))
+        .pipe(
+            tap((v) => {
+                console.debug('client-api.APISUB', 'post graphistry-subscribe', 'exor');
+                g.iFrame.contentWindow.postMessage({ type: 'graphistry-subscribe', agent: 'graphistryjs', path: SELECTION_PATH }, '*');
+            }),
+            finalize(() => {
+                console.debug('client-api.APISUB', 'finalize post graphistry-unsubscribe', 'exor');
+                g.iFrame.contentWindow.postMessage({ type: 'graphistry-unsubscribe', agent: 'graphistryjs', path: SELECTION_PATH }, '*');
+            }),
+            switchMap(() =>
+                fromEvent(window, 'message').pipe(
+                    map(o => o.data),
+                    tap(o => console.log('client-api.APISUB', 'recieved message', o, 'exor')),
+                    filter(o => o && o.type === 'graphistry-update' && o.path === SELECTION_PATH),
+                    map(o => o.data),
+                    tap(({ edge, point}) => {
+                        g.models.model.setCache({
```

Comment:
don't need a subscribe?

Reply:
no, I checked it works to do a read.

Also, setCache returns the model, not an observable.